checks/handlerCheck.go

```diff
@@ -0,0 +1,123 @@
+)
+
+type handlerRuntime func(HandlerConfigs) error
+
+type HandlerConfigs struct {
+	handlerName string
+	conf        *config.Configuration
+}
+
+var handlerCheck = map[string]handlerRuntime{
+	"dotnet": dotnet,
+	"java":   java,
+	"node":   node,
+	"python": python,
+}
+
+var NEW_RELIC_LAMBDA_WRAPPER = "newrelic|lambda|wrapper.handler"
+var handlerPath = "/var/task"
+
+func checkHandler(conf *config.Configuration, reg *api.RegistrationResponse) error {
+	functionHandler := reg.Handler
+	fmt.Println(functionHandler)
+	fmt.Println("hello from check handler")
+	for k, v := range handlerCheck {
+		err := checkRuntime(k)
+		if err == nil {
+			h := HandlerConfigs{
+				handlerName: functionHandler,
+				conf:        conf,
+			}
+			err = v(h)
+			if err != nil {
+				return fmt.Errorf("Lambda handler is set incorrectly: %s", err)
+			}
+			break
+		}
+	}
+	// If we make it here that means the runtime is
+	// custom and we don't want to throw an error
+	return nil
+}
+
+func checkRuntime(runtime string) error {
+	path := fmt.Sprintf("/var/lang/bin/%s", runtime)
+	return pathExists(path)
+}
+
+// Handler format for dotnet functions: Assembly::Namespace.ClassName::Method
+// from the file names alone all that we can validate is the Assembly dll
+func dotnet(h HandlerConfigs) error {
+	path := pathFormatter(h.handlerName, "dll")
+	return pathExists(path)
+}
+
+// Handler format for java functions: package.Class::method
+// we can validate that the package + class path exist
+// and want to validate that the method value is set to interface method handleRequest
+func java(h HandlerConfigs) error {
+	if !strings.HasSuffix(h.handlerName, "handleRequest") {
+		return fmt.Errorf("Function handler missing required 'handleRequest' method")
+	}
+	path := pathFormatter(h.handlerName, "class")
+	return pathExists(path)
+}
+
+// Hander format for node functions: file.function
+// we can validate that the handler has been set to newrelic-lambda-wrapper.handler"
+// and that the file name provided in the env var NEW_RELIC_LAMBDA_HANDLER exists
+func node(h HandlerConfigs) error {
+	functionHandler := checkNrHandler(h, "-")
+	path := pathFormatter(functionHandler, "js")
+	return pathExists(path)
+}
+
+// Hander format for python functions: file.function
+// we can validate that the handler has been set to newrelic_lambda_wrapper.handler
+// and that the file name provided in the env var NEW_RELIC_LAMBDA_HANDLER exists
+func python(h HandlerConfigs) error {
+	functionHandler := checkNrHandler(h, "_")
+	path := pathFormatter(functionHandler, "py")
+	return pathExists(path)
+}
+
+func checkNrHandler(h HandlerConfigs, dividingChar string) string {
+	wrapper := strings.ReplaceAll(NEW_RELIC_LAMBDA_WRAPPER, "|", dividingChar)
+	if h.handlerName != wrapper {
+		return h.handlerName
+	}
+	return *h.conf.NRHandler
+}
+
+func pathFormatter(functionHandler string, fileType string) string {
+	re := regexp.MustCompile("[::.]+")
```

Comment:
Why use a regular expression instead of using `path`? If there a case here that the latter didn't handle?

https://golang.org/pkg/path/

Reply:
> Why use a regular expression instead of using `path`? If there a case here that the latter didn't handle?
> 
> https://golang.org/pkg/path/

the path package implements utility routines for manipulating slash-separated paths which may not always be the case for our handlers. The regex is for formatting the handler into a structure that allows for file existence checking.